src/resizing.jl

```diff
@@ -137,11 +141,30 @@ function imresize{T,N}(original::AbstractArray{T,N}, short_size::NTuple)
     imresize(original, new_size)
 end
 
+"""
+    imresize(img, sz) -> imgr
+
+Change `img` to be of size `sz`. This interpolates the values at
+sub-pixel locations. If you are shrinking the image, you risk aliasing
+unless you low-pass filter `img` first. For example:
+
+    σ = map((o,n)->0.75*o/n, size(img), sz)
+    kern = KernelFactors.gaussian(σ)   # from ImageFiltering
+    imgr = imresize(imfilter(img, kern, NA()), sz)
+
+See also [`restrict`](@ref).
+"""
 function imresize{T,N}(original::AbstractArray{T,N}, new_size::NTuple{N})
+    # To choose the output type, rather than forcing everything to
+    # Float64 by multiplying by 1.0, we exploit the fact that the scale
+    # changes correspond to integer ratios.  We mimic ratio arithmetic
+    # without actually using Rational (which risks promoting to a
+    # Rational type, too slow for image processing).
+    Tnew = typeof((first(original) * 1) / 1)
```

Comment:
Hmm, yes, since those "mimic" floating-point (meaning, they round automatically) that might be better. Good catch.